### DISCUSSION TITLE
Guide to Open Source

### DISCUSSION THREAD

**Discussion by dominicduffin1** [Virtual-Coffee/virtualcoffee.io#419] [Open Source]
I am making a draft PR with an outline for the Guide to Open Source, would be keen to hear the team's thoughts on what else I should include or if anything included should be out of scope here.

I've also marked a couple of areas where I would like some input.

See draft PR #420

**Comment by danieltott** [comment]
This is great @dominicduffin1 !

We recently re-organized the resources section a bit, and now have index pages for each section, including the open source articles: https://virtualcoffee.io/resources/open-source/

It'd be great to spruce that up a bit!

As far as the outline goes - you have some really good ideas cooking in there. I really like `Funding and governance in open source` - _definitely_ interested in reading that one 😄 

For `Contributors checklist`: I at first was thinking this was a checklist for contributing, but "how to evaluate an open source project" would be a really useful article. Thinking about a contributors checklist made me want to go and make sure we have "tl;dr" checklists for pretty much everything we put out 😂 

Anyway - very excited about the direction this is going 🎉

**Comment by BekahHW** [comment]
Maybe sections for best practices for both maintainers and contributors would be nice. And I could see for that section for maintainers having a subsection on creating community and avoiding burnout.

**Comment by dominicduffin1** [comment]
@BekahHW I love the idea of having a subsection on creating community and avoiding burnout!

**Comment by dominicduffin1** [comment]
I have pushed what I've written for this, it may still want a bit of editing and maybe there's some more things we could add, I'm not sure?

#420 

https://deploy-preview-420--virtual-coffee-io.netlify.app/resources/open-source/
https://deploy-preview-420--virtual-coffee-io.netlify.app/resources/open-source/guide-to-os/

I'm wondering how would be best to organize it and incorporate the existing content - should we have several pages or one Guide to Open Source page - what are your thoughts on this @danieltott @BekahHW ?

I'd like to see in there the Career Building with Open Source section but not sure I'm qualified to write that part.

Also guide to Gitlab, I've never used it but it would probably be good to have this, especially as Hacktoberfest started supporting Gitlab in 2021.

**Reply by BekahHW** [reply]
@dominicduffin1 this is awesome!!!

I would say anything you think should belong but don't feel like you can add to the doc can be an issue after the resource is up. 

I'm going back and forth on whether or not it should be one guide or several pages. 

I have to take a deeper look at everything, but thank you so much for putting this together.

I'd love for @adiati98 to chime in if she has any input as well.

**Reply by adiati98** [reply]
@dominicduffin1 this looks great!

I don't have time to take a look at it deeper today, but I took a glimpse at it.
I will take a look at it deeper later on.

I think that we can make this into several pages on [Open Source Resource](https://virtualcoffee.io/resources/open-source/).

Based on your update, I'm thinking that we can have a Maintainer Guide, a Contributor Guide, and something like a Git & GitHub 101 page.
We can always add GitLab later on when it's preferable/necessary.

As for the Hacktoberfest section, I'm not sure if we want to keep it under the OSS resource page since it's a yearly basis occasion.
Because we have [Virtual Coffee: Hacktoberfest Initiative](https://hacktoberfest.virtualcoffee.io/), I'm thinking to include this section in there.

Any thought about this, @dominicduffin1 @BekahHW @danieltott?

**Reply by dominicduffin1** [reply]
@BekahHW @adiati98 

I think the Maintainer Guide, Contributor Guide and Git & GitHub 101 pages sounds good. Maybe another page called About Open Source or something similar? I'm thinking that the existing Maintainer Checklist page could fit in both Maintainer Guide and Contributor Guide, we could look at whether each could have a version oriented toward maintainers or contributors specifically or have the same for both (which could be retained as a separate page, idk?).

Adding things I don't think I can do as issues after it is up sounds good!

**Reply by adiati98** [reply]
@dominicduffin1 Sorry it takes me a while!
If you have time some time next week, we can meetup and discuss this 😊

**Reply by adiati98** [reply]
Hi @BekahHW @danieltott,

@dominicduffin1 and I had a discussion about this.
And we are thinking to have following pages and sections to update our Open Source Resources:

1. **About Open Source**
  - What is open-source
    **Good things to know:**
  - Funding and governance in open source
  - How to use open source to build a resume

2. **Git & GitHub 101**
  - How to use git & GitHub

3. **Contributor Guide**
  - Writing issues (we will move the existing page here)
  - Best practice for contributors

4. **Maintainer Guide**
  - Maintainer checklist (we will move the existing page here)
  - Best practices for maintainers
  - Repository checklist
  - Open-sourcing an existing project

We also have a section about Hacktoberfest. And we're wondering if we can put this in the [Virtual Coffee: Hacktoberfest Initiative](https://hacktoberfest.virtualcoffee.io/) because Hacktoberfest is a yearly event?
If we can put this on the Hacktoberfest page, we already have some Q&A there.
So, should we follow the Q&A format for this section (and update some of the answers), or create a new section for that?

Any thoughts on all of this?

Thank you in advance 😊
